### PR TITLE
Align snapshot restore shard size estimation condition

### PR DESCRIPTION
**Pull request by idegtiarenko**
This pr aligns snapshot restore shard size estimation condition with internal one in SnapshotShardSizeInfo::getShardSize

### CODE REVIEW

**Review by kingherc — APPROVED** [review]
LGTM.

**Comment by idegtiarenko** [comment]
@elasticsearchmachine please run elasticsearch-ci/part-3

File: server/src/main/java/org/elasticsearch/snapshots/SnapshotShardSizeInfo.java [hunk 1]
```diff
@@ -24,11 +25,8 @@ public SnapshotShardSizeInfo(Map<InternalSnapshotsInfoService.SnapshotShard, Lon
     }
 
     public Long getShardSize(ShardRouting shardRouting) {
-        if (shardRouting.primary()
-            && shardRouting.active() == false
-            && shardRouting.recoverySource().getType() == RecoverySource.Type.SNAPSHOT) {
-            final RecoverySource.SnapshotRecoverySource snapshotRecoverySource = (RecoverySource.SnapshotRecoverySource) shardRouting
-                .recoverySource();
+        if (shardRouting.active() == false && shardRouting.recoverySource().getType() == RecoverySource.Type.SNAPSHOT) {
+            final SnapshotRecoverySource snapshotRecoverySource = (SnapshotRecoverySource) shardRouting.recoverySource();
```
Comment (DaveCTurner) [inline server/src/main/java/org/elasticsearch/snapshots/SnapshotShardSizeInfo.java:29]: Could we `assert shardRouting.primary()` here?

**Review by DaveCTurner — APPROVED** [review]
One nit about an assertion otherwise LGTM

**Pull request merged**